File: task-03/prime.hs
isPrime :: Int -> Bool
isPrime num
  | num < 2 = False
  | num == 2 || num == 3 = True
  | num `mod` 2 == 0 || num `mod` 3 == 0 = False
  | otherwise = checkPrime 5
  where
    checkPrime i
      | i * i <= num = num `mod` i /= 0 && num `mod` (i + 2) /= 0 && checkPrime (i + 6)
      | otherwise = True

getInput :: IO Int
getInput = do
  putStr "Type a number: "
  input <- getLine
  let k = read input :: Int
  if k >= 2 then
    return k
  else
    getInput

printPrimes :: Int -> IO ()
printPrimes n = mapM_ (\x -> putStrLn (show x)) (filter isPrime [2..n])

main :: IO ()
main = do
  k <- getInput
  printPrimes k

File: task-03/prime.java
import java.util.Scanner;

public class Prime {

    public static boolean isPrime(int num) {
        if (num < 2) return false;
        if (num == 2 || num == 3) return true;
        if (num % 2 == 0 || num % 3 == 0) return false;

        int i = 5;
        while (i * i <= num) {
            if (num % i == 0 || num % (i + 2) == 0) return false;
            i += 6;
        }

        return true;
    }

    public static int getInput() {
        Scanner scanner = new Scanner(System.in);
        int k = 0;
        while (k < 2) {
            System.out.print("Type a number: ");
            k = scanner.nextInt();
        }
        scanner.close();
        return k;
    }

    public static void printPrimes(int n) {
        for (int i = 2; i <= n; i++) {
            if (isPrime(i)) System.out.println(i);
        }
    }

    public static void main(String[] args) {
        int k = getInput();
        printPrimes(k);
    }
}
